security/fma-incident-response-improvements.md

```diff
@@ -0,0 +1,254 @@
+    - [Mitigations](#mitigations-1)
+    - [Detection](#detection-1)
+    - [Recovery Path(s)](#recovery-paths-1)
+    - [Action items](#action-items)
+  - [FM3: Anchor state fails to progress within a time bound](#fm3-anchor-state-fails-to-progress-within-a-time-bound)
+    - [Description](#description-2)
+    - [Risk Assessment](#risk-assessment-2)
+      - [Mitigations](#mitigations-2)
+      - [Detection](#detection-2)
+      - [Recovery Path(s)](#recovery-paths-2)
+      - [Action items:](#action-items-1)
+  - [FM4: Errors in bond distribution](#fm4-errors-in-bond-distribution)
+    - [Description](#description-3)
+    - [Risk Assessment](#risk-assessment-3)
+    - [Mitigations](#mitigations-3)
+    - [Detection](#detection-3)
+    - [Recovery Path(s)](#recovery-paths-3)
+    - [Action items](#action-items-2)
+    - [Generic items we need to take into account:](#generic-items-we-need-to-take-into-account)
+  - [Action Items](#action-items-3)
+  - [Audit Requirements](#audit-requirements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+_Italics are used to indicate things that need to be replaced._
+
+|                    |                          |
+| ------------------ | ------------------------ |
+| Author             | wildmolasses             |
+| Created at         | 2025-01-22               |
+| Initial Reviewers  | smartcontracts           |
+| Need Approval From | Matt Solomon, Josep Bové |
+| Status             | In Review                |
+
+## Introduction
+
+This document covers the fault dispute game incident response improvements project. The project makes modifications to several key contracts to improve incident response capabilities:
+
+ - **FaultDisputeGame**: Has changes but almost any failure mode here is equivalent to a bug in the dispute game which we already have runbooks for. The main new consideration is potential accounting issues in the new bond refund functionality (documented in [FM4](#fm4-errors-in-bond-distribution)).
+
+- **DelayedWETH**: Contract is basically unchanged and the diff is so minor there's really no genuine concern for any sort of failure.
+
+- **AnchorStateRegistry**: Since this change doesn't actually make the ASR a critical dependency yet (except for `isGameRespected`), the only real failure modes are:
+  - The anchor state not being updated correctly (documented in [FM3](#fm3-anchor-state-fails-to-progress-within-a-time-bound))
+  - Incorrect anchor state being set
+  - (We're already planning to add monitoring for anchor state getting too old and we can easily add monitoring for anchor state being invalid)
+
+- **OptimismPortal**: Very minor changes, with two main failure modes:
+  - The new incident response functionality being misused (documented in [FM2](#fm2-legacy-runbookresponse-confusion))
+  - A critical issue that allows a game that isn't respected to be used (either a bug in the changes or a bug in `ASR.isGameRespected`)
```

Comment:
Should there be a failure mode for this?

Reply:
thanks, added in c19e884